admin/category_product_listing.php

```diff
@@ -397,11 +401,11 @@
             $zc_skip_products = false;
             $zc_skip_categories = false;
 
-            if ($check_products == 0) {
+            if ($check_products === 0) {
               $zc_skip_products = false;
               $zc_skip_categories = false;
             }
-            if ($check_categories == true) {
+            if ($check_categories) {
```

Comment:
$check_categories is a (bool) return from `zen_has_category_subcategories`; use `=== true`.